### PR TITLE
SOA proposal reply

### LINKED ISSUE #54: [SOA_bot] add yes/no responses to proposed orders
response system to the proposals

if proposer = allies, response. if not allies then no response

### PR DESCRIPTION
**Pull request by wwongkamjan**
resolve #54

### CODE REVIEW

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 1]
```diff
@@ -79,6 +79,24 @@ def gen_messages(self, orders_list: List[str]):
 
         return msgs_data
 
+    def gen_proposal_reply(self, best_proposer: str, prp_orders: dict, messages: MessagesData):
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:82]: Please add return type annotation

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 2]
```diff
@@ -101,6 +119,9 @@ def __call__(self, rcvd_messages: List[Tuple[int, Message]]):
         orders_data = OrdersData()
         orders_data.add_orders(best_orders)
 
-        # generate messages
+        # generate messages for FCT sharing info orders
         messages = self.gen_messages(orders_data)
+
+        # generate proposal response YES/NO to allies
+        messages = self.gen_proposal_reply(best_proposer, prp_orders, messages)
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:126]: since gen_proposal_reply doesnt seem to return anything, doesnt messages get set to None?

**Comment by trigaten** [comment]
Also, if you use a specific keyword in your comment, like
resolves #54, the issue gets automatically closed when this is merged.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 2 above)
Comment (wwongkamjan) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:127]: I forgot to return MessageData. thanks for reviewing!

**Comment by wwongkamjan** [comment]
resolves https://github.com/ALLAN-DIP/baseline_bots/issues/54

I have resolved the review comments. Could anyone review this again? :)

**Pull request merged**